### PR TITLE
Add pkcs11 provider.

### PR DESCRIPTION
**Pull request by petedyerarm**
Added the PKCS11 provider as an alternative to the SW TPM provider.

The provider to be used is controlled using the PARSEC_PROVIDER variable in local.conf

This can either be SOFTWARE_TPM or PKCS11.

The relevant files will then be selected based upon that.

### CODE REVIEW

**Comment by petedyerarm** [comment]
Merge with
https://github.com/PelionIoT/meta-parsec/pull/16
https://github.com/PelionIoT/meta-pelion-edge/pull/332

File: conf/local.conf [hunk 1]
```diff
@@ -110,4 +110,7 @@ PREFERRED_RPROVIDER_kubelet = "kubelet"
 VOLATILE_LOG_DIR = "no"
 
 # Set default Pelion image name to mfg tool
-MFGTOOL_FLASH_IMAGE = "console-image-lmp"
\ No newline at end of file
+MFGTOOL_FLASH_IMAGE = "console-image-lmp"
+
+# Set default for the PARSEC Provider to be SOFTWARE_TPM.
```
Comment (kivaisan) [inline conf/local.conf:115]: Should we have the options listed in the comment? (`SOFTWARE_TPM` and `PKCS11`)

Comment (JanneKiiskila) [reply conf/local.conf:115]: PKCS11 should be the default, as quickly as we can get it working on all of the targets.
SW TPM is really just a stop-gap.
HW TPM should be the good secondary choice.

Comment (JanneKiiskila) [reply conf/local.conf:115]: I suppose in theory you could have actually multiple ones, i.e. PKCS11 via OPTEE and ALSO a HW TPM.

**Review by JanneKiiskila — APPROVED** [review]

**Pull request merged**